File: tests/test_interpreter.py
from unittest import TestCase
from vsi.ast import AST, AssignStatement, Integer
from vsi.interpreter import Interpreter


class TestInterpreter(TestCase):

    def test___repr___method(self):
        env = {}
        interpreter = Interpreter(AST([AssignStatement("x", 10)]), env)
        interpreter_repr = repr(interpreter)
        expected_repr = "Interpreter(Statements([AssignStatement(x, 10)]), {})"
        self.assertEqual(interpreter_repr, expected_repr)

    def test_interpret_method(self):
        env = {}
        ast = AST([AssignStatement("x", Integer(10))])
        interpreter = Interpreter(ast, env)
        interpreter.interpret()
        self.assertIn("x", env)
        self.assertEqual(env["x"], 10)

File: Vsi.Tests/TestInterpreter.cs
using System.Collections.Generic;
using Xunit;
using Vsi.Core;

namespace Vsi.Tests
{
    public class TestInterpreter
    {
        [Fact]
        public void TestInterpreterReprMethod()
        {
            var env = new Dictionary<string, object?>();
            var interpreter = new Interpreter(new Ast(new List<AstNode> { new AssignStatement("x", new Integer(10)) }), env);
            Assert.Equal("Interpreter(Statements([AssignStatement(x, Integer(10))]), {})", interpreter.ToString());
        }

        [Fact]
        public void TestInterpretMethod()
        {
            var env = new Dictionary<string, object?>();
            var ast = new Ast(new List<AstNode> { new AssignStatement("x", new Integer(10)) });
            var interpreter = new Interpreter(ast, env);
            interpreter.Interpret();
            Assert.True(env.ContainsKey("x"));
            Assert.Equal(10, env["x"]);
        }
    }
}
